src/compiler/transformers/classFields.ts

```diff
@@ -1050,14 +1276,65 @@ namespace ts {
                     }
                 }
             }
-            if (shouldTransformPrivateElementsOrClassStaticBlocks &&
-                isPrivateIdentifierInExpression(node)) {
+            if (isPrivateIdentifierInExpression(node)) {
                 // #x in obj
                 return transformPrivateIdentifierInInExpression(node);
             }
             return visitEachChild(node, visitor, context);
         }
 
+        function visitCommaListExpression(node: CommaListExpression, discarded: boolean) {
+            const elements = discarded ?
+                visitCommaListElements(node.elements, discardedValueVisitor) :
+                visitCommaListElements(node.elements, visitor, discardedValueVisitor);
+            return factory.updateCommaListExpression(node, elements);
+        }
+
+        function visitParenthesizedExpression(node: ParenthesizedExpression, discarded: boolean, referencedName: Expression | undefined) {
+            // 8.4.5 RS: NamedEvaluation
+            //   ParenthesizedExpression : `(` Expression `)`
+            //     ...
+            //     2. Return ? NamedEvaluation of |Expression| with argument _name_.
+
+            const visitorFunc: Visitor =
+                discarded ? discardedValueVisitor :
+                referencedName ? node => namedEvaluationVisitor(node, referencedName) :
+                visitor;
+            const expression = visitNode(node.expression, visitorFunc, isExpression);

```

Comment:
`referencedName` seems to be passed through for nearly every node kind. Is it possible do something like visitorFunc for every node kind in classFields.ts and cut down on the number of visit functions that need to be overridden?

Reply:
not without adding additional branching for the two different `updateX` functions, plus variable declarations don't have a `dotDotDotToken` property or a `propertyName`, so I'd have to add just as much code to handle differences between the two nodes as I do with two separate functions, and the separate functions are easier to read.

Reply:
The regular `visitor` function is passed significantly more often than `namedEvaluationVisitor`, since named evaluation only occurs in a small number of cases. An assigned name is only produced when performing an assignment to an `Identifier` or a property/field, and only carries through `ParenthesizedExpression` (and our own `PartiallyEmittedExpression`, which is only used to preserve comments), and only applies to `ClassExpression` in this transformer.

I'd rather not add checks for a `referencedName` that will always be undefined, and unused branches that would produce unused closures due to the potential runtime cost and overhead.